resolver/src/main/java/io/netty/resolver/SimpleNameResolver.java

```diff
@@ -168,12 +156,81 @@ public final boolean isResolved(SocketAddress address) {
         }
     }
 
+
+    @Override
+    public final Future<List<T>> resolveAll(String inetHost, int inetPort) {
+        return resolveAll(InetSocketAddress.createUnresolved(checkNotNull(inetHost, "inetHost"), inetPort));
+    }
+
+    @Override
+    public Future<List<T>> resolveAll(String inetHost, int inetPort, Promise<List<T>> promise) {
+        return resolveAll(InetSocketAddress.createUnresolved(checkNotNull(inetHost, "inetHost"), inetPort), promise);
+    }
+
+    @Override
+    public final Future<List<T>> resolveAll(SocketAddress address) {
+        if (!isSupported(checkNotNull(address, "address"))) {
+            // Address type not supported by the resolver
+            return executor().newFailedFuture(new UnsupportedAddressTypeException());
+        }
+
+        if (isResolved(address)) {
```

Comment:
We didn't have it. It was only fixed in `DnsNameResolver`. I'm not sure if it's a good place to put a fix though, because a resolver implementation might be using similar-looking string with some other meaning.